### PR TITLE
Intelligent error display on plugin

### PR DESCRIPTION
**Pull request by jusasiiv**
Verbose error handling

### CODE REVIEW

File: blockonomics-woocommerce.php [hunk 1]
```diff
@@ -156,29 +156,62 @@ public function process_payment($order_id)
...
-                try {
-                    $blockonomics = new Blockonomics;
-                    $address = $blockonomics->new_address(get_option('blockonomics_api_key'), get_option("blockonomics_callback_secret"));
-                    $price = $blockonomics->get_price(get_woocommerce_currency());
-                } catch (Exception $e) {
-                    $address = '';
+                $blockonomics = new Blockonomics;
+                $responseObj = $blockonomics->new_address(get_option('blockonomics_api_key'), get_option("blockonomics_callback_secret"));
+                $price = $blockonomics->get_price(get_woocommerce_currency());
+
```
Comment (shivaenigma) [inline blockonomics-woocommerce.php]: Would be good if the following error code is refactored into a new method, for cleaner code

**Comment by shivaenigma** [comment]
All string must be put into translation construct so that these can be [translated](https://translate.wordpress.org/projects/wp-plugins/blockonomics-bitcoin-payments) into other languages, example
`_('Sorry, but there was an error processing your order. Please try again or try a different payment method. (plugin not configured)', 'blockonomics-bitcoin-payments')`

**Comment by blockonomics** [comment]
Tested.  Look Ok. Please update condition to detect in case of non response obj
`
403.. API key, 503 ...blocking connection... other errors print http status code`

**Comment by jusasiiv** [comment]
The response code parsing added some complexity, but if that is fine it is easy now to add conditions on internal server errors / http response errors. 

`$responseObj->{'response_code'} = $http_response_header[0];` was added because if the response was empty, the whole object was empty and could not be parsed for message

**Pull request merged**